test/k8sT/Policies.go

```diff
@@ -591,6 +605,82 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
 			testConnectivitytoRedis()
 		})
 	})
+
+	Context("KubernetesNetworkPolicy between server and client", func() {
+		BeforeEach(func() {
+			kubectl = helpers.CreateKubectl(helpers.K8s1VMName(), logger)
+
+			By("Creating the namespace that will be used for the pods")
+			_, err := kubectl.CoreV1().Namespaces().Create(&v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}})
+			Expect(err).NotTo(HaveOccurred())
+
+			By("Creating a simple server that serves on port 80 and 81.")
+			podServer, service = createServerPodAndService(kubectl, namespace, "server", []int{80, 81})
+
+			By("Waiting for pod ready", func() {
+				ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+				defer cancel()
+				err := kubectl.WaitForPodReady(ctx, namespace, podServer.Name)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			By("Testing pods can connect to both ports when no policy is present.")
+			testCanConnect(kubectl, namespace, "client-can-connect-80", service, 80, true)
+			testCanConnect(kubectl, namespace, "client-can-connect-81", service, 81, true)
+		})
+
+		AfterEach(func() {
+			cleanupServerPodAndService(kubectl, podServer, service)
+			By("Deleting the namespace that was used for the pods")
+			err := kubectl.CoreV1().Namespaces().Delete(namespace, &metav1.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			if CurrentGinkgoTestDescription().Failed {
+				ciliumPod, _ := kubectl.GetCiliumPodOnNode(helpers.KubeSystemNamespace, helpers.K8s1)
+				kubectl.CiliumReport(helpers.KubeSystemNamespace, ciliumPod, []string{
+					"cilium bpf tunnel list",
+					"cilium endpoint list"})
+			}
+		})
+
+		It("should enforce policy based on NamespaceSelector", func() {
+			if helpers.GetCurrentK8SEnv() == "1.6" {
+				Skip("K8s 1.6 doesn't offer a default deny mode in its policy")
+				return
+			}
+
+			policy := &networkingv1.NetworkPolicy{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "allow-ns-b-via-namespace-selector",
+				},
+				Spec: networkingv1.NetworkPolicySpec{
+					// Apply to server
+					PodSelector: metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"pod-name": podServer.Name,
+						},
+					},
+					// Allow traffic only from NS-B
+					Ingress: []networkingv1.NetworkPolicyIngressRule{{
+						From: []networkingv1.NetworkPolicyPeer{{
+							NamespaceSelector: &metav1.LabelSelector{
+								MatchLabels: map[string]string{
+									"ns-name": namespace,
+								},
+							},
+						}},
+					}},
+				},
+			}
+			policy, err := kubectl.NetworkingV1().NetworkPolicies(namespace).Create(policy)
+			Expect(err).NotTo(HaveOccurred())
+			defer cleanupNetworkPolicy(kubectl, policy)
```

Comment:
Why not `kubectl.CiliumPolicyAction(`?

Reply:
The policy is not a file

Reply:
Then, that policy helper should have something that all pods are updated with that policy. If not we'll have race conditions as we had in the past. Why don't you create the policy in a file? It's easy, can be valid for multiple version and it's the standard way of using it.

Reply:
@eloycoto we could have such policies as variables which are accessed by a variety of tests. This allows us to purely test against the Kubernetes API as opposed to having to add more code around importing of files locally. I prefer what @aanm is doing here.